python-frontend/src/test/java/org/sonar/python/index/VariableDescriptorTest.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * SonarQube Python Plugin
+ * Copyright (C) 2011-2021 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.python.index;
+
+
+import java.util.Collection;
+import org.junit.Test;
+import org.sonar.plugins.python.api.symbols.Symbol;
+import org.sonar.python.tree.TreeUtils;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.sonar.python.PythonTestUtils.lastExpression;
+import static org.sonar.python.index.DescriptorUtils.descriptors;
+
+public class VariableDescriptorTest {
+
+  @Test
+  public void variableDescriptor() {
+    VariableDescriptor x = lastVariableDescriptor(
+      "x: int = 42",
+      "x");
+    // only typeshed symbols have it != null
```

Comment:
Do we want that to change eventually? Or have a way to retrieve the information anyway?

Probably we can consider generalizing the approach that in TypeShed a type annotation is a type certainty and say that we never question type annotation when they happen outside of the file under analysis? Wdyt?

Reply:
I think we could eventually do this, however as discussed let's focus on having something that works as it is today. I think we can discuss about this improvement later